pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -90,7 +144,11 @@ func (rs *ReceiverService) GetReceivers(ctx context.Context, q models.GetReceive
 		return nil, err
 	}
 
-	// TODO: check for list access
+	eval := accesscontrol.EvalPermission(accesscontrol.ActionAlertingReceiversList)
+	listAccess, err := rs.ac.Evaluate(ctx, user, eval)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I would recommend using an approach with authz service (see RuleService) and check whether user has access to every contact point. In future, you will need to just update the authz service and not the code here.

Also, if I am not mistaken, this code is used in provisioning. Therefore, this check automatically affects it. I think we should be very careful adding new permissions to provisioning, and do this when we have all CRUD operations covered.

Reply:
Remember the `:list` permission will not be a scoped permission so we'll only ever check for org level access. My current plan is to use a service for all of the scoped permissions (`:read`, `secrets:read` (?), `:delete`), wdyt?

This permission being checked in provisioning has already bitten me in tests, do you propose we revert the provisioning service for now?

Reply:
we can remove authz logic from service for now and protect only endpoint.

Reply:
I missed the fact that it determines what output to return to user. I think this is ok now.